src/get-started/java-binding.md

```diff
@@ -0,0 +1,379 @@
+systems. A set of operations is defined by each service and can include sending
+transactions to the network, read requests for the blockchain data with the
+provision of the cryptographic proof, etc. Exonum provides an embedded web
+framework for describing the REST-interface of the service.
+
+`com.exonum.binding.service.Service#createPublicApiHandlers` method is used to
+determine the handlers for HTTP requests. These handlers are available at the
+common path corresponding to the service name. Thus, the `/balance/:walletId`
+handler for balance requests in the "cryptocurrency" service will be available
+at `/cryptocurrency/balance/:walletId`.
+
+See [documentation][vertx.io] on the possibilities of `Vert.x` used as a web
+framework.
+
+### Dependencies Management
+
+To describe the dependencies of the service components (both system-specific
+ones, for example, Exonum time service, and external ones), Guice is used. Each
+service should define a module describing implementations of the framework
+interfaces - `com.exonum.binding.service.Service`,
+`com.exonum.binding.service.TransactionConverter` and other components
+implementions, if any.
+
+!!! note "Minimalistic Example of Service Module"
+    ```java
+    public class ServiceModule extends AbstractModule {
+
+      @Override
+      protected void configure() {
+        // Define the Service implementation.
+        bind(Service.class).to(CryptocurrencyService.class).in(Singleton.class);
+
+        // Define the TransactionConverter implementation.
+        bind(TransactionConverter.class).to(CryptocurrencyTransactionConverter.class);
+  }
+}
+```
+The full name of the module class must be submitted when the service launches.
+
+For more information on using Guice, see the [project wiki][Guice].
+
+### Testing
+
+#### Of Schema and Manipulations with the Storage
+
+To test the schema and manipulations with the storage, Exonum provides a
+database that stores the values in the RAM -
+`com.exonum.binding.storage.database.MemoryDb`. Before using it in integration
+tests, it is necessary to load a library with the implementation of native
+methods:
+
+```java
+public class MySchemaIntegrationTest {
+  static {
+    LibraryLoader.load();
+  }
+
+  // Tests
+}
+```
+
+The plug-in for running the tests should be configured to transmit
+`java.library.path` parameter to JVM:
+
+```
+-Djava.library.path=<path-to-java-bindings-library>
+```
+
+`Surefire/Failsafe` for Maven should be configured as follows:
+
+```xml
+<plugin>
+    <!-- You can also configure a failsafe to run integration tests during
+         'validate' phase of a Maven build to separate unit tests and ITs. -->
+    <artifactId>maven-surefire-plugin</artifactId>
+    <configuration>
+        <argLine>
+            -Djava.library.path=${path-to-java-bindings-library}
+        </argLine>
+    </configuration>
+</plugin>
+```
+
+Each test can create `Snapshot`, `Fork`, using `MemoryDb`, and also apply `Fork`
+to its state:
```

Comment:
Each test can create a `Snapshot`and/or `Fork`, using `MemoryDb`

Also, it's not very clear what 'its' refers to, so perhaps better to indicate the object to remove ambiguity.